doc/hardware/peripherals/opamp.rst

```diff
@@ -0,0 +1,28 @@
+.. _opamp_api:
+
+Operational Amplifier (OPAMP)
+#############################
+
+Overview
+********
+
+The operational amplifier is an analog device that amplifies differential input
+signals (difference between inverting and non-inverting input) to give a resulting
+output voltage.
+
+
+Configuration
+*************
+
+When internal OPAMP is enabled, an initial configuration must be provided using
+the devicetree. The OPAMP can configure non-inverting or inverting gain at runtime
+using device driver API.
```

Comment:
```suggestion
When OPAMP is enabled, an initial configuration shall be provided using
the devicetree. The OPAMP gain can be adjusted at runtime.
```

- it does not matter in general if it is internal or not
- `shall` - could be more appropriate I think
- gain is programmable for sure, but by programming it, it will be adjusted ;)
- it does not matter if gain is inverting or not in this context
- it does not matter if gain is set by applying device driver API